orchestrator/workflow.py

```diff
@@ -90,7 +96,8 @@ class Workflow(Protocol):
     __qualname__: str
     name: str
     description: str
-    authorize_callback: Callable[[OIDCUserModel | None], bool]
+    authorize_callback: Authorizer
+    retry_auth_callback: Authorizer
```

Comment:
I can see different arguments for what to name these. For now, I've chosen to not change Alex's naming of `authorize_callback` in case it's already in use downstream, but I'm open to feedback on how best to name the following callbacks. Each is a function implementing a RBAC policy.

* `authorize_callback`: called when a user starts a workflow.
* `resume_auth_callback`: called when a user resumes a workflow paused by an `inputstep`
* `retry_auth_callback`:  called when a failed step is retried. Arg to both `@workflow` and `@inputstep`.